DivanteLtd/vue-storefront
# add flag is_server_method when replacing payment methods

### Short description and why it's useful

This pr fixes two errors:
A. https://github.com/DivanteLtd/vue-storefront/commit/6e418b027a6dfacd4e820b3c18a669f58466ea27
We can't make order second time because we don't add `is_server_method` when replacing payment methods in `cart/overrideServerTotals`. Then in `cart/syncPaymentMethods` it treat backend methods as added on client side (like in this module `src/modules/payment-cash-on-delivery/index.ts`).
QA:
1. go to any product page and add it to your cart
2. go through shopping path (go through: Personal Details, Shipping, Payment, Review order)
3. click "Place the order" button
4. again go to any product page and add it to your cart
5. go through shopping path (go through: Personal Details, Shipping, Payment, Review order)
6. click "Place the order" button

B. https://github.com/DivanteLtd/vue-storefront/commit/8e98bc87b713678450275f72fff026327ed4a1d2
Payment details is not updated in vuex, only in component. So when we change country we make request to get payment methods for specific country. And for this request we use state from vuex, which is not updated on fly. I've added action to update payment details in vuex when country is changed.
QA:
1. go to any product page and add it to your cart
2. go through shopping path to Payment
3.a. change country and check in devtools request `/api/cart/shipping-information` - in billingAddress there should be selected `countryId`
3.b. switch between Afghanistan and Albania - payment methods should change

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)


### Upgrade Notes and Changelog


**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance
